linkerd/drain/src/lib.rs

```diff
@@ -88,14 +92,19 @@ impl Watch {
         F: FnOnce(&mut A),
     {
         tokio::select! {
-            res = &mut future => return res,
-            _ = self.rx => {
+            res = &mut future => res,
+            _handle = self.handle() => {
                 on_drain(&mut future);
-                (&mut future).await
+                future.await
             }
         }
     }
 
+    pub async fn handle(self) -> Handle {
```

Comment:
TIOLI: would be nice to have a comment explaining what this does exactly, because we now have three options for consuming the `Watch`.